documentation/proposals/Proposal - Multi-Backend Input.md

```diff
@@ -257,27 +265,25 @@ This will be configurable on `Mice` (i.e. via `InputContext.Mice.ClickConfigurat
-public readonly record struct MouseClickEvent(IMouse Mouse, Vector2 Position, MouseButton Button);
-public readonly record struct JoystickDownEvent(IJoystick Joystick, JoystickButton Button);
-public readonly record struct JoystickUpEvent(IJoystick Joystick, JoystickButton Button);
-public readonly record struct JoystickHatMoveEvent(IJoystick, Vector2 Value, Vector2 Delta);
-public readonly record struct JoystickAxisMoveEvent(IJoystick Joystick, int Axis, float Value, float Delta);
-public readonly record struct GamepadDownEvent(IGamepad Gamepad, JoystickButton Button);
-public readonly record struct GamepadUpEvent(IGamepad Gamepad, JoystickButton Button);
-public readonly record struct GamepadThumbstickMoveEvent(IJoystick, Vector2 Value, Vector2 Delta);
-public readonly record struct GamepadTriggerMoveEvent(IJoystick Joystick, int Axis, float Value, float Delta);
+public readonly record struct ConnectionEvent(IInputDevice Device, DateTime Timestamp, bool IsConnected);
+public readonly record struct KeyDownEvent(IKeyboard Keyboard, DateTime Timestamp, Key Key, bool IsRepeat);
+public readonly record struct KeyUpEvent(IKeyboard Keyboard, DateTime Timestamp, Key Key);
+public readonly record struct KeyCharEvent(IKeyboard Keyboard, DateTime Timestamp, char Character);
+public readonly record struct MouseDownEvent(IMouse Mouse, DateTime Timestamp, Vector2 Position, MouseButton Button);
+public readonly record struct MouseUpEvent(IMouse Mouse, DateTime Timestamp, Vector2 Position, MouseButton Button);
+public readonly record struct MouseMoveEvent(IMouse Mouse, DateTime Timestamp, Vector2 Position, Vector2 Delta);
+public readonly record struct MouseScrollEvent(IMouse Mouse, DateTime Timestamp, Vector2 Position, Vector2 WheelPosition, Vector2 Delta);
+public readonly record struct MouseClickEvent(IMouse Mouse, DateTime Timestamp, Vector2 Position, MouseButton Button);
+public readonly record struct JoystickDownEvent(IJoystick Joystick, DateTime Timestamp, JoystickButton Button);
+public readonly record struct JoystickUpEvent(IJoystick Joystick, DateTime Timestamp, JoystickButton Button);
+public readonly record struct JoystickHatMoveEvent(IJoystick Joystick, DateTime Timestamp, Vector2 Value, Vector2 Delta);
+public readonly record struct JoystickAxisMoveEvent(IJoystick Joystick, DateTime Timestamp, int Axis, float Value, float Delta);
+public readonly record struct GamepadDownEvent(IGamepad Gamepad, DateTime Timestamp, JoystickButton Button);
+public readonly record struct GamepadUpEvent(IGamepad Gamepad, DateTime Timestamp, JoystickButton Button);
+public readonly record struct GamepadThumbstickMoveEvent(IJoystick Joystick, DateTime Timestamp, Vector2 Value, Vector2 Delta);
+public readonly record struct GamepadTriggerMoveEvent(IJoystick Joystick, DateTime Timestamp, int Axis, float Value, float Delta);
 ```
 
-This is the part of this proposal that incorporates the ideas in Enhanced Input Events, and is why this proposal supersedes that one.
-
```

Comment:
This is important background information, please do not remove.

Reply:
i know, but i think this is really out of place because it's on the middle of the reading. You see, i don't deleted all the information and there's the credits of the Enhanced Input Events proposal in the start of the file
You think we can just move it or something? Part of my proposal is bellow and it can cause misunderstanding

Reply:
The Working Group disagrees given they have approved this information in its place. I don’t think this is what will cause misunderstanding.

Reply:
If you're saying, ok